mcs/class/System/Test/System.Net.Security/SslStreamTest.cs

```diff
@@ -93,6 +171,167 @@ void AuthenticateClientAndServer (bool server, bool client)
 		}
 	}
 
+	[Category("NotWorking")]    // Works in .Net, doesn't work in mono.
+	[Test()]
+	public void SelfCompatibilityTest()
+	{
+		#if __MonoCS__
```

Comment:
That should not be a problem as Mono.Security.dll is already required to
build System.dll unit tests.

> TEST_MCS_FLAGS = -r:System.Drawing.dll -r:Mono.Security.dll
-r:System.Data -r:System.Xml.dll -r:System.Core.dll -
> nowarn:618,672,219,67,169,612 \
> $(foreach f, $(TEST_RESOURCES), -resource:$(f),$(notdir $(f)))

OTOH using the define constant means you're checking which compiler (csc vs
mcs) was used, not the runtime platform - and that's incorrect.

Sebastien

On Tue, Jan 6, 2015 at 10:41 AM, Edward Ned Harvey <notifications@github.com
> wrote:

> In mcs/class/System/Test/System.Net.Security/SslStreamTest.cs
> <https://github.com/mono/mono/pull/1490#discussion-diff-22529030>:
>
> > @@ -93,6 +171,167 @@ void AuthenticateClientAndServer (bool server, bool client)
> >  		}
> >  	}
> >
> > +	[Category("NotWorking")]    // Works in .Net, doesn't work in mono.
> > +	[Test()]
> > +	public void SelfCompatibilityTest()
> > +	{
> > +		#if __MonoCS__
>
> The way to programatically add or remove root or intermediate certs with
> mono requires Mono.Security.X509. The way to do it in .Net requires
> System.Security.Cryptography.X509Certificates. Sure, we *could* copy
> Mono.Security.dll to the windows project and eliminate the use of #if
> MonoCS in favor of a runtime check that causes Windows to avoid calling
> mono Mono.Security.X509StoreManager.CurrentUser.TrustedRoot.Import(), but
> we would be literally pointlessly adding the Mono.Security assembly
> reference to a .Net project that doesn't use it, just to eliminate the #if.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mono/mono/pull/1490/files#r22529030>.
>